app/results.py

```diff
@@ -206,10 +206,14 @@ def get(self):
                     return self.redirect(create_url)
 
         if self.params.role == 'seek':
+            # The query_name parameter was previously called "query", and we
+            # continue to support it for third-parties that link directly to
+            # search results pages.
+            if self.params.query:
+                self.params.query_name = self.params.query
```

Comment:
Would be better to avoid rewriting self.params directly, which is usually considered immutable, though the existing code was written in that way.

Also, nitpick, but `query_name` should probably win (because it's the canonical one) if both `query_name` and `query` are specified.

So it should be:
`query_name = self.params.query_name || self.params.query`
and use `query_name` instead of `self.params.query_name` below.

Reply:
Done.